api/compose/api.go

```diff
@@ -177,4 +183,24 @@ type Stack struct {
 // LogConsumer is a callback to process log messages from services
 type LogConsumer interface {
 	Log(service, container, message string)
+	Status(service, container, msg string)
+}
+
+// ContainerEventListener is a callback to process ContainerEvent from services
+type ContainerEventListener func(event ContainerEvent)
+
+// ContainerEvent notify an event has been collected on Source container implementing Service
+type ContainerEvent struct {
+	Type     int
+	Source   string
+	Service  string
+	Line     string
```

Comment:
Should we have more typed logs (stdout/err) ? (Will be useful to add timestamps also later on)

Reply:
Would be nice, and would also address https://github.com/docker/compose/issues/8098
`stdcopy.StdCopy` we use could distinguish container's stdout vs stderr, but I'm not sure we can do the same for containers configured to set a TTY